codegen/src/api/custom_values.rs

```diff
@@ -47,14 +47,15 @@ fn generate_custom_value_fn(
     if fn_names_taken.contains(&fn_name) {
         return None;
     }
-    let fn_name_ident = format_ident!("{fn_name}");
+    // if the fn_name would be an invalid ident, return None:
```

Comment:
I do wonder, would this happen when the function name is comming from the chain metadata (substrate/polkadot)?
I believe the `ident` parsing would fail when the input is not a valid variable name or keyword.
We could probably take this approach everywhere else in the codegen, since I guess `Ident::new` would panic for those cases too.

Reply:
For custom values I wanted to be extra sure, because people can put in there whatever they want. I think the rest of the metadata is at least valid in terms of identifiers, because they are already used as identifiers on the substrate side, where the metadata is created. But we could definitely use `syn::parse_str::<syn::Ident>` instead of `format_ident!` in a few places.

Reply:
It's a good point that we should only use `format_ident` when we are confident that the value must be an ident :) Good catch changing this to `parse_str` here!